include/seqan3/range/view/translation.hpp

```diff
@@ -45,292 +45,657 @@
+    REV_FRAME_0 = 1 << 3,                               //!< The first reverse frame starting at position 0
+    REV_FRAME_1 = 1 << 4,                               //!< The second reverse frame starting at position 1
+    REV_FRAME_2 = 1 << 5,                               //!< The third reverse frame starting at position 2
+    FWD_REV_0 = FWD_FRAME_0 | REV_FRAME_0,              //!< The first forward and first reverse frame
+    FWD_REV_1 = FWD_FRAME_1 | REV_FRAME_1,              //!< The second forward and second reverse frame
+    FWD_REV_2 = FWD_FRAME_2 | REV_FRAME_2,              //!< The first third and third reverse frame
+    FWD = FWD_FRAME_0 | FWD_FRAME_1 | FWD_FRAME_2,      //!< All forward frames
+    REV = REV_FRAME_0 | REV_FRAME_1 | REV_FRAME_2,      //!< All reverse frames
+    SIX_FRAME = FWD | REV                               //!< All frames
 };
 
+//!\brief Enable bitwise operators for enum translation_frames.
+template<>
+constexpr bool add_enum_bitwise_operators<translation_frames> = true;
 }
 
 namespace seqan3::detail
 {
-/*!\brief Lambda for translation of one nucleotide triplet into a single aminoacid.
- * \details Used by view::translate_single.
- */
-auto translate = [] (auto range)
-{
-    return translate_triplet(range);
-};
 
-/*!\brief The underlying type of seqan3::view::translate_single.
+/*!\brief The return type of seqan3::view::translate_single.
+ * \implements seqan3::view_concept
+ * \implements seqan3::sized_range_concept
+ * \implements seqan3::random_access_range_concept
+ * \tparam urng_t The type of the range being translated.
+ * \param tf Translation frame to be used (translation_frames::FWD_FRAME_0,
+ * translation_frames::FWD_FRAME_1, translation_frames::FWD_FRAME_2,
+ * translation_frames::REV_FRAME_0,translation_frames::REV_FRAME_1,
+ * translation_frames::REV_FRAME_2).
  * \ingroup view
  */
-struct translate_single_fn
+template <typename urng_t>
+//!\cond
+    requires sized_range_concept<urng_t> &&
+             random_access_range_concept<urng_t> &&
+             nucleotide_concept<std::decay_t<reference_t<std::decay_t<urng_t>>>>
+//!\endcond
+class view_translate_single
 {
-    /*!\brief Single frame translation from nucleotide to aminoacid.
-     * \tparam irng_t The type of the range being processed.
-     * \param irange The range being processed.
+private:
+    //!\brief The data members of view_translate_single.
+    struct data_members_t
+    {
+        //!\brief The input range (of ranges).
+        urng_t urange;
+        //!\brief The frame that should be used for translation.
+        translation_frames const tf;
+    };
+    //!\brief Storage of data members.
+    std::shared_ptr<data_members_t> data_members;
+
+    //!\brief Error thrown if tried to be used with multiple frames.
+    static constexpr char multiple_frame_error[] = "Error: Invalid type of frame. Choose one out of FWD_FRAME_0, REV_FRAME_0, FWD_FRAME_1, REV_FRAME_1, FWD_FRAME_2 and REV_FRAME_2.";
```

Comment:
You can use our `constexpr_string` for this now, also please check the maximum line width.